source/MaterialXRender/Util.cpp

```diff
@@ -231,51 +238,90 @@ unsigned int getUIProperties(ConstValueElementPtr nodeDefElement, UIProperties&
     return propertyCount;
 }
 
-unsigned int getUIProperties(const string& path, DocumentPtr doc, const string& target, UIProperties& uiProperties)
+void createUIPropertyGroups(DocumentPtr doc, const VariableBlock& block, UIPropertyGroup& groups,
+                            UIPropertyGroup& unnamedGroups, const string& pathSeparator, bool showAllInputs)
 {
-    ValueElementPtr valueElement = findNodeDefChild(path, doc, target);
-    if (valueElement)
+    // Generated an ordered map of shader inputs.
+    using ShaderInputPair = std::pair<InputPtr, ShaderPort*>;
+    std::map<int, ShaderInputPair> shaderInputMap;
+    for (ShaderPort* variable : block.getVariableOrder())
     {
-        return getUIProperties(valueElement, uiProperties);
-    }
-    return 0;
-}
+        if (!variable->getValue())
```

Comment:
I'm not quite sure we want to prune these out. e.g. I think this will prune out empty filenames on image nodes.

Reply:
In this case, I believe getValue() will only return nullptr if no value at all is bound to the shader port, so we should be safe in the case of an empty filename.

Reply:
I think this goes towards another issue logged that input filenames which have no value are not showing up in the UI.
As I'm not sure yet if this is the cause of pruning vs pruning of uniforms curing code gen what you have is the same logic as before just a better early out. All looks good. Thanks.